src/application/service_tester.go

```diff
@@ -141,11 +144,46 @@ func (s *ServiceTester) setup() error {
 
 // Run runs the tests for the service and return true if the tests passed
 // and an error if any
-func (s *ServiceTester) Run() (int, error) {
+func (s *ServiceTester) Run() (types.TestResult, error) {
 	if err := s.setup(); err != nil {
-		return 1, err
+		return types.TestResult{
+			Passed:      false,
+			Interrupted: false,
+		}, err
+	}
+
+	var isInterrupted bool
+	var err error
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		signal.Stop(c)
+		err = s.Shutdown()
+		isInterrupted = true
+		wg.Done()
+	}()
+	if isInterrupted {
+		return types.TestResult{
+			Passed:      false,
```

Comment:
Booleans default to false so this could be `types.TestResult{Interrupted: true}`. Please apply this in similar places elsewhere.